### PR TITLE
Fix small error in page row count limiting

### PR DESCRIPTION
**Pull request by etseidl**
When page row limit is a multiple of chunk size, then an extra chunk is written per page (eg if max_page_size_rows is 20000, 25000 will actually be written). This change fixes that.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by PointKernel** [comment]
ok to test

**Comment by PointKernel** [comment]
ok to test

**Review by PointKernel — APPROVED** [review]
LGTM

**Comment by davidwendt** [comment]
Is it possible to write a simple test case for this?

**Review by nvdbaranec — APPROVED** [review]

**Comment by etseidl** [comment]
> Is it possible to write a simple test case for this?

Sure, I'll do that.

**Comment by etseidl** [comment]
@davidwendt that proved harder than I thought it would be, but I added a test that writes a file with a 5000 row per page limit, and then parse the first data page header to make sure only 5000 values were written.

**Review by davidwendt — APPROVED** [review]
Cool. 100 lines of test source for testing 1 line of code. I like it. Nice work.

File: cpp/tests/io/parquet_test.cpp [hunk 1]
```diff
@@ -3323,4 +3323,109 @@ TEST_F(ParquetWriterTest, EmptyListWithStruct)
   cudf::test::expect_columns_equal(result.tbl->view().column(0), *L0);
 }
 
+class ThriftReaderLite {
```
Comment (PointKernel) [inline cpp/tests/io/parquet_test.cpp:3326]: We have a thrift reader here https://github.com/rapidsai/cudf/blob/branch-22.08/cpp/src/io/parquet/compact_protocol_reader.hpp. Can you please include the header `#include <src/io/parquet/compact_protocol_reader.hpp>` in this file and use that one instead?

**Review by PointKernel — CHANGES_REQUESTED** [review]
@etseidl One final suggestion to clean up the test code.

File: cpp/tests/io/parquet_test.cpp (hunk 1 above)
Comment (etseidl) [reply cpp/tests/io/parquet_test.cpp:3326]: @PointKernel  will do.  Yes, I stole that from compact_protocol_reader anyway, but wasn't sure I could include that in the test code.

Comment (etseidl) [reply cpp/tests/io/parquet_test.cpp]: done!

**Review by PointKernel — APPROVED** [review]

**Pull request merged**